src/lib.rs

```diff
@@ -100,6 +101,17 @@ cfg_if! {
     }
 }
 
+macro_rules! verify_event_key {
+    ($event:expr) => {{
+        if $event.key == usize::MAX {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "the key is not allowed to be `usize::MAX`",
+            ));
+        }
+    }};
+}
+
 /// Key associated with notifications.
 const NOTIFY_KEY: usize = std::usize::MAX;
```

Comment:
I think this needs a `#[cfg(...)]` to get around the warnings.

Reply:
@stjepang Which warnings are you talking about? I can't seem to find any in the (horrible) GitHub checks UI.